sdk/communication/azure-communication-identity/src/main/java/com/azure/communication/identity/CommunicationIdentityAsyncClient.java

```diff
@@ -253,21 +318,54 @@ public Mono<AccessToken> getToken(CommunicationUserIdentifier communicationUser,
     }
 
     /**
-     * Gets a token for an identity with response.
+     * Gets a Communication Identity access token for a {@link CommunicationUserIdentifier}.
      *
-     * @param communicationUser The user to be issued tokens.
-     * @param scopes The scopes that the token should have.
-     * @return The access token with response.
+     * @param communicationUser A {@link CommunicationUserIdentifier} from whom to issue a Communication Identity
+     * access token.
+     * @param scopes List of {@link CommunicationTokenScope} scopes for the Communication Identity access token.
+     * @return the Communication Identity access token.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Response<AccessToken>> getTokenWithResponse(CommunicationUserIdentifier communicationUser,
+    public Mono<AccessToken> getToken(CommunicationUserIdentifier communicationUser,
         Iterable<CommunicationTokenScope> scopes) {
+        try {
+            Objects.requireNonNull(communicationUser);
+            Objects.requireNonNull(scopes);
+            return getToken(communicationUser, scopes, null);
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Gets a Communication Identity access token for a {@link CommunicationUserIdentifier}.
+     *
+     * @param communicationUser A {@link CommunicationUserIdentifier} from whom to issue a Communication Identity
+     * access token.
+     * @param scopes List of {@link CommunicationTokenScope} scopes for the Communication Identity access token.
+     * @param tokenExpiresAfter Custom validity period of the Communication Identity access token within &lt;60,1440&gt;
+     * minutes range. If not provided, the default value of 1440 minutes (24 hours) will be used.
+     * @return the Communication Identity access token with response.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<AccessToken>> getTokenWithResponse(CommunicationUserIdentifier communicationUser,
+        Iterable<CommunicationTokenScope> scopes, Duration tokenExpiresAfter) {
         try {
             Objects.requireNonNull(communicationUser);
             Objects.requireNonNull(scopes);
             final List<CommunicationTokenScope> scopesInput = StreamSupport.stream(scopes.spliterator(), false).collect(Collectors.toList());
+
```

Comment:
creation of the CommunicationIdentityAccessTokenRequest is kind of duplicated in several places, maybe it's better to have a private method for it?

Reply:
Does it make sense to encapsulate one line in a private method?

Reply:
The following lines are identical:
CommunicationIdentityAsyncClient
129 -> 142 & 188 -> 203
321 -> 337 & 388 -> 405

CommunicationIdentityAsyncClient
110 -> 118 & 151 -> 159
257 -> 264 & 310 -> 317

We can create `util` package where you can create a package private final `CommunicationIdentityClientUtils` class  where you can define static methods for creating instance of the CommunicationIdentityCreateRequest, CommunicationIdentityAccessTokenRequest and also move the getTokenExpirationInMinutes method & OVERFLOW_MESSAGE constant to that utils class, so that you can access to these methods from CommunicationIdentityClient, CommunicationIdentityClientAsync and test classes instead of duplicating. It will be easier for us in terms of maintenance

Reply:
Done.